platform/darwin/src/NSExpression+MGLAdditions.mm

```diff
@@ -1097,7 +1097,23 @@ - (id)mgl_jsonExpressionObject {
                 NSArray *arguments = self.arguments.mgl_jsonExpressionObject;
                 return [@[@"concat", self.operand.mgl_jsonExpressionObject] arrayByAddingObjectsFromArray:arguments];
             } else if ([function isEqualToString:@"objectFrom:withIndex:"]) {
-                return @[@"at", self.arguments[1].mgl_jsonExpressionObject, self.arguments[0].mgl_jsonExpressionObject];
+                id index = self.arguments[1].mgl_jsonExpressionObject;
+                
+                if ([self.arguments[1] expressionType] == NSConstantValueExpressionType
+                    && [[self.arguments[1] constantValue] isKindOfClass:[NSString class]]) {
+                    id value = self.arguments[1].constantValue;
+                    
+                    if ([value isEqualToString:@"FIRST"]) {
+                        index = [NSExpression expressionForConstantValue:@0].mgl_jsonExpressionObject;
+                    } else if ([value isEqualToString:@"LAST"]) {
+                        NSExpression *count = [NSExpression expressionWithFormat:@"count(%@) - 1", self.arguments[0]];
+                        index = count.mgl_jsonExpressionObject;
```

Comment:
Consider inlining `count` here, because the variable name `count` is a bit misleading. (It’s the last index, not the count.)